### PR TITLE
asyncio: add tests for exceptions in coroutines

### PR DESCRIPTION
**Pull request by galdor**

### CODE REVIEW

File: tests/contrib/asyncio/test_tracer.py [hunk 1]
```diff
@@ -85,3 +85,69 @@ def test_context_task_none(self):
...
         ctx = self.tracer.get_call_context()
         ok_(ctx is not None)
+
+    @mark_asyncio
+    def test_exception(self):
+        async def f1():
+            with self.tracer.trace('f1'):
+                raise Exception('f1 error')
+
+        try:
```
Comment (palazzem) [inline tests/contrib/asyncio/test_tracer.py]: ``self.assertRaises`` should already handle the exception (without a re-raise). You can remove the ``try-finally`` block in these tests.

File: tests/contrib/asyncio/test_tracer.py [hunk 2]
```diff
@@ -85,3 +85,69 @@ def test_context_task_none(self):
...
+            spans = traces[0]
+            eq_(1, len(spans))
+            span = spans[0]
+            eq_(1, span.error)
+
+    @mark_asyncio
+    def test_nested_exceptions(self):
+        async def f1():
+            with self.tracer.trace('f1'):
+                raise Exception('f1 error')
```
Comment (palazzem) [inline tests/contrib/asyncio/test_tracer.py]: can you also add a test for the ``aiohttp``? it's enough that a test raises an exception like ``1/0`` without handling. You can check it there: https://github.com/DataDog/dd-trace-py/blob/palazzem/async-python/tests/contrib/aiohttp/test_middleware.py#L40

Also can you check the error message and the stack trace?
```python
span.get_tag('error.msg')
# 'f1 error'
span.get_tag('error.stack')
# 'Traceback (most recent call last):\n  File ".../tests/contrib/asyncio/test_tracer.py", line 93, in f1\n    raise Exception(\'f1 error\')\nException: f1 error\n'
```
for the stack you can use something like:
```python
ok_('Exception: f1 error' in span.get_tag('error.stack'))
```

File: tests/contrib/aiohttp/app/web.py [hunk 3]
```diff
@@ -23,6 +23,11 @@
     span.finish()
     return web.Response(text=text)
 
+async def route_exception(request):
+    raise Exception('error')
+
+async def route_async_exception(request):
```
Comment (palazzem) [inline tests/contrib/aiohttp/app/web.py]: what's the difference with ``route_exception``?

File: tests/contrib/aiohttp/test_middleware.py [hunk 4]
```diff
@@ -151,3 +151,33 @@ class TestTraceMiddleware(AioHTTPTestCase):
...
+        await request.text()
+
+        traces = self.tracer.writer.pop_traces()
+        eq_(1, len(traces))
+        spans = traces[0]
+        eq_(1, len(spans))
+        span = spans[0]
+        eq_(1, span.error)
+        eq_('error', span.get_tag('error.msg'))
+        ok_('Exception: error' in span.get_tag('error.stack'))
```
Comment (palazzem) [inline tests/contrib/aiohttp/test_middleware.py]: can you also check if the resource is properly set? something like: https://github.com/DataDog/dd-trace-py/blob/894550e97763d8b2fd702422962d109322ccb2c8/tests/contrib/aiohttp/test_middleware.py#L75

File: tests/contrib/aiohttp/app/web.py (hunk 3 above)
Comment (galdor) [inline tests/contrib/aiohttp/app/web.py]: oops.

**Pull request merged**